lib/tasks/migrations/migrations.rake

```diff
@@ -2,28 +2,31 @@
 
 namespace :migrations do
   COMMON = {
-    headers: { "Content-Type" => "application/json" }
+    headers: { "Content-Type" => "application/json" },
+    batch_size: 1000,
   }.freeze
 
   desc "Migrates v2 resources to v3"
   task :roles, [] => :environment do |_task, _args|
     has_encountred_issue = 0
 
-    Role.select(:id, :name).where.not(name: Role::RESERVED_ROLE_NAMES).each do |r|
-      params = { role: { name: r.name } }
-      response = Net::HTTP.post(uri('roles'), payload(params), COMMON[:headers])
-
-      case response
-      when Net::HTTPCreated
-        puts green "Succesfully migrated Role:"
-        puts cyan "  ID: #{r.id}"
-        puts cyan "  Name: #{r.name}"
-      else
-        puts red "Unable to migrate Role:"
-        puts yellow "  ID: #{r.id}"
-        puts yellow "  Name: #{r.name}"
-        has_encountred_issue = 1 # At least one of the migrations failed.
-      end
+    Role.select(:id, :name)
+        .where.not(name: Role::RESERVED_ROLE_NAMES)
+        .find_each(batch_size: COMMON[:batch_size]) do |r|
```

Comment:
If we're going to use the default, I would drop the explicit `batch_size: ..` to explicitly show we're using the default value

Reply:
I've added it to enable customization to suit admins needs without having to know about the internal of `ActiveRecord`,
You wanna me to remove it?

Reply:
ummm no that's a fair point. it's fine as is